File: ios/StableChannels/StableChannels/Views/Settings/StablePositionView.swift
import SwiftUI
import UIKit

struct StablePositionView: View {
    @Environment(AppState.self) private var appState
    @State private var copiedCounterparty = false

    var body: some View {
        List {
            Section {
                HStack {
                    Text(String(localized: "label_expected_usd", defaultValue: "Expected USD"))
                    Spacer()
                    Text(appState.stableChannel.expectedUSD.formatted)
                        .fontWeight(.medium)
                }
                HStack {
                    Text(String(localized: "label_backing_sats", defaultValue: "Backing Sats"))
                    Spacer()
                    Text(appState.stableChannel.backingSats.satsFormatted)
                }
                HStack {
                    Text(String(localized: "label_native_btc", defaultValue: "Native BTC"))
                    Spacer()
                    Text(appState.stableChannel.nativeChannelBTC.sats.satsFormatted)
                }

                if appState.btcPrice > 0 && appState.stableChannel.expectedUSD.amount > 0 {
                    let result = StabilityService.checkStabilityAction(
                        appState.stableChannel, price: appState.btcPrice
                    )
                    HStack {
                        Text(String(localized: "label_status", defaultValue: "Status"))
                        Spacer()
                        Text(result.action.rawValue)
                            .foregroundStyle(stabilityColor(result.action))
                            .fontWeight(.medium)
                    }
                    HStack {
                        Text(String(localized: "label_distance_from_par", defaultValue: "Distance from Par"))
                        Spacer()
                        Text(String(format: "%.2f%%", result.percentFromPar))
                            .foregroundStyle(result.percentFromPar < 0.1 ? .green : .orange)
                    }
                }

                if !appState.stableChannel.counterparty.isEmpty {
                    Button {
                        UIPasteboard.general.string = appState.stableChannel.counterparty
                        withAnimation { copiedCounterparty = true }
                        DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
                            withAnimation { copiedCounterparty = false }
                        }
                    } label: {
                        HStack {
                            Text(String(localized: "label_counterparty", defaultValue: "Counterparty"))
                                .foregroundStyle(.primary)
                            Spacer()
                            if copiedCounterparty {
                                Label(
                                    String(localized: "button_copied", defaultValue: "Copied"),
                                    systemImage: "checkmark"
                                )
                                .font(.caption)
                                .foregroundStyle(.green)
                                .transition(.scale.combined(with: .opacity))
                            } else {
                                Text(String(appState.stableChannel.counterparty.prefix(8)) + "..."
                                    + String(appState.stableChannel.counterparty.suffix(8)))
                                    .font(.system(.caption, design: .monospaced))
                                    .foregroundStyle(.secondary)
                            }
                        }
                    }
                }
            }
        }
        .navigationTitle(String(localized: "title_stable_position", defaultValue: "Stable Position"))
        .navigationBarTitleDisplayMode(.inline)
    }

    private func stabilityColor(_ action: StabilityService.StabilityAction) -> Color {
        switch action {
        case .stable: return .green
        case .pay: return .orange
        case .checkOnly: return .blue
        case .highRiskNoAction: return .red
        }
    }
}

File: android/app/src/main/java/com/stablechannels/app/ui/settings/StablePositionView.kt
package com.stablechannels.app.ui.settings

import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalClipboardManager
import androidx.compose.ui.text.AnnotatedString
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.stablechannels.app.AppState
import com.stablechannels.app.services.StabilityService
import com.stablechannels.app.util.satsFormatted

@Composable
fun StablePositionView(appState: AppState) {
    val sc by appState.stableChannel.collectAsState()
    val btcPrice by appState.priceService.currentPrice.collectAsState()
    val clipboardManager = LocalClipboardManager.current
    var copiedCounterparty by remember { mutableStateOf(false) }

    val stabilityResult = remember(sc, btcPrice) {
        StabilityService.checkStabilityAction(sc, btcPrice)
    }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(16.dp)
    ) {
        if (sc.expectedUSD.amount > 0) {
            // Expected USD — prominent
            SettingsDetailRow(
                label = "Expected USD",
                value = sc.expectedUSD.formatted,
                valueColor = Color(0xFF10B981),
                valueBold = true
            )
            Spacer(Modifier.height(16.dp))

            // Backing Sats
            SettingsDetailRow(
                label = "Backing Sats",
                value = sc.backingSats.satsFormatted()
            )
            Spacer(Modifier.height(16.dp))

            // Native BTC
            SettingsDetailRow(
                label = "Native BTC",
                value = sc.nativeChannelBTC.formatted,
                valueColor = Color(0xFFF59E0B)
            )
            Spacer(Modifier.height(16.dp))

            // Stability Status — colored
            val statusColor = when (stabilityResult.action.value) {
                "STABLE" -> Color(0xFF10B981)
                "PAY" -> Color(0xFFF59E0B)
                "CHECK_ONLY" -> Color(0xFF3B82F6)
                "HIGH_RISK_NO_ACTION" -> Color(0xFFEF4444)
                else -> MaterialTheme.colorScheme.onSurface
            }
            SettingsDetailRow(
                label = "Status",
                value = stabilityResult.action.value,
                valueColor = statusColor,
                valueBold = true
            )
            Spacer(Modifier.height(16.dp))

            // Distance from Par — colored
            if (stabilityResult.percentFromPar > 0) {
                val parColor = if (stabilityResult.percentFromPar < 0.1) Color(0xFF10B981) else Color(0xFFF59E0B)
                SettingsDetailRow(
                    label = "Distance from Par",
                    value = String.format("%.2f%%", stabilityResult.percentFromPar),
                    valueColor = parColor
                )
                Spacer(Modifier.height(16.dp))
            }

            // Counterparty — tap to copy
            val cpk = sc.counterparty
            if (cpk.isNotEmpty()) {
                Surface(
                    onClick = {
                        clipboardManager.setText(AnnotatedString(cpk))
                        copiedCounterparty = true
                    },
                    shape = MaterialTheme.shapes.medium,
                    tonalElevation = 1.dp,
                    modifier = Modifier.fillMaxWidth()
                ) {
                    Row(
                        modifier = Modifier.padding(16.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Column(modifier = Modifier.weight(1f)) {
                            Text(
                                text = "Counterparty",
                                style = MaterialTheme.typography.bodyMedium,
                                color = MaterialTheme.colorScheme.onSurfaceVariant
                            )
                            Spacer(Modifier.height(4.dp))
                            Text(
                                text = "${cpk.take(8)}...${cpk.takeLast(8)}",
                                style = MaterialTheme.typography.bodyMedium,
                                fontFamily = FontFamily.Monospace
                            )
                        }
                        Text(
                            text = if (copiedCounterparty) "Copied ✓" else "Tap to copy",
                            style = MaterialTheme.typography.labelSmall,
                            color = if (copiedCounterparty) Color(0xFF10B981) else MaterialTheme.colorScheme.onSurfaceVariant
                        )
                    }
                }
            }
        } else {
            Text(
                text = "No stable position active",
                style = MaterialTheme.typography.bodyLarge,
                color = MaterialTheme.colorScheme.onSurfaceVariant
            )
            Spacer(Modifier.height(8.dp))
            Text(
                text = "Trade BTC to USD to create a stable position.",
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurfaceVariant
            )
        }
    }
}

@Composable
private fun SettingsDetailRow(
    label: String,
    value: String,
    valueColor: Color = Color.Unspecified,
    valueBold: Boolean = false
) {
    Row(
        modifier = Modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically
    ) {
        Text(
            text = label,
            style = MaterialTheme.typography.bodyLarge,
            color = MaterialTheme.colorScheme.onSurface
        )
        Text(
            text = value,
            style = MaterialTheme.typography.bodyLarge,
            color = if (valueColor != Color.Unspecified) valueColor else MaterialTheme.colorScheme.onSurface,
            fontWeight = if (valueBold) FontWeight.SemiBold else FontWeight.Normal
        )
    }
}
